project/src/com/google/daggerquery/executor/models/Query.java

```diff
@@ -46,6 +47,13 @@
   private final static String ALLPATHS_QUERY_NAME = "allpaths";
   private final static String SOMEPATH_QUERY_NAME = "somepath";
 
+  private final static int MAX_NUMBER_OF_MISPLACED_LETTERS = 3;
```

Comment:
Out of curiosity, how did you settle on a maximum of 3?

Reply:
I thought that this is not too small and not too many number of fixes. Do you think it makes sense to make it more or less? 🦄

Reply:
My gut is that it might need to be more, but I have no data to back this up. So let's leave it for now, and change it if we see any potential problems cropping up.

For more context, I'm a terrible speller. The most common mistake that I make is inversion (eg. thsi  instead of this). Using the LevenshteinDistance an inversion would mean that two letters are misplaced. So using the example of "thsi" above, LevenshteinDistance would say that "his" is a closer match than "this" although while typing I feel like the latter may be just as likely an error.

... But as you can see, this opinion is entirely anecdotal. So unless we feel like linguistically analyzing typing mistakes, I think that your initial number is a good place to start.